WalletWasabi.Fluent/Models/WebcamQrReader.cs

```diff
@@ -0,0 +1,164 @@
+
+		public event EventHandler<string>? InvalidAddressFound;
+
+		public event EventHandler<Exception>? ErrorOccured;
+
+		private AsyncLock ScanningTaskLock { get; set; }
+		public bool RequestEnd { get; set; }
+		public Network Network { get; }
+		public Task? ScanningTask { get; set; }
+		public bool IsRunning => ScanningTask is not null;
+
+		public async Task StartScanningAsync()
+		{
+			using (await ScanningTaskLock.LockAsync().ConfigureAwait(false))
+			{
+				if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+				{
+					ErrorOccured?.Invoke(this, new NotImplementedException("This operating system is not supported."));
+				}
+				ScanningTask = Task.Run(() =>
+				{
+					VideoCapture? camera = null;
+					try
+					{
+						camera = new();
+						camera.SetExceptionMode(true);
+						// Setting VideoCaptureAPI to DirectShow, to remove warning logs,
+						// might need to be changed in the future for other operating systems
+						if (!camera.Open(DefaultCameraId, VideoCaptureAPIs.DSHOW))
+						{
+							throw new InvalidOperationException("Could not open webcamera.");
+						}
+						RequestEnd = false;
+						KeepScan(camera);
+					}
+					catch (Exception exc)
+					{
+						Logger.LogError("QR scanning stopped. Reason:", exc);
+						ErrorOccured?.Invoke(this, exc);
+					}
+					finally
+					{
+						camera?.Release();
+						camera?.Dispose();
+					}
+				});
+			}
+		}
+
+		public async Task StopScanningAsync()
+		{
+			using (await ScanningTaskLock.LockAsync().ConfigureAwait(false))
+			{
+				if (ScanningTask is { } task)
+				{
+					RequestEnd = true;
+					await task;
+
+					ScanningTask = null;
+				}
+			}
+		}
+
+		private void KeepScan(VideoCapture camera)
+		{
+			PixelSize pixelSize = new(camera.FrameWidth, camera.FrameHeight);
+			Vector dpi = new(96, 96);
+			using WriteableBitmap writeableBitmap = new(pixelSize, dpi, PixelFormat.Rgba8888, AlphaFormat.Unpremul);
+
+			int dataSize = camera.FrameWidth * camera.FrameHeight;
+			int[] helperArray = new int[dataSize];
+			using QRCodeDetector qRCodeDetector = new();
+			using Mat frame = new();
+			while (!RequestEnd)
+			{
+				try
+				{
+					bool gotBackFrame = camera.Read(frame);
+					if (!gotBackFrame || frame.Width == 0 || frame.Height == 0)
+					{
+						continue;
+					}
+					ConvertMatToWriteableBitmap(frame, writeableBitmap, helperArray);
+
+					NewImageArrived?.Invoke(this, writeableBitmap);
+
+					if (qRCodeDetector.Detect(frame, out Point2f[] points))
+					{
+						using Mat tmpMat = new();
+						string qrCode = qRCodeDetector.Decode(frame, points, tmpMat);
+						if (string.IsNullOrWhiteSpace(qrCode))
+						{
+							continue;
+						}
+						if (AddressStringParser.TryParse(qrCode, Network, out _))
+						{
+							CorrectAddressFound?.Invoke(this, qrCode);
+							break;
+						}
+						else
+						{
+							InvalidAddressFound?.Invoke(this, qrCode);
+						}
+					}
+				}
+				catch (OpenCVException exc)
```

Comment:
nit: I think that people mostly use:

```suggestion
				catch (OpenCVException ex)
```

or

```suggestion
				catch (OpenCVException e)
```

`exc` is new.